message L4SysProtoMsg {
	// parameters
	required int32 instr_offset = 1;
	required int32 bit_offset = 2;

	// results
	// make these optional to reduce overhead for server->client communication
	enum ResultType {
		DONE = 1;
		TIMEOUT = 2;
		TRAP = 3;
		INTR = 4;
		WRONG = 5;
		UNKNOWN = 6;
	}
	// instruction pointer where injection was done
	optional uint32 injection_ip = 3;
	// result type, see above
	optional ResultType resulttype = 4;
	// result data, depending on resulttype (see source code)
	optional uint32 resultdata = 5;
	// generated output
	optional string output = 6;
	// optional textual description of what happened
	optional string details = 7;
}
